cmake/thirdparty/SetupCUDA.cmake

```diff
@@ -7,6 +7,15 @@
 # Sanity Checks
 ################################
 
+# Ensure CUDA_TOOLKIT_ROOT_DIR is specified, since it is needed 
+# by the call to find_packag(CUDA) below.
+if (NOT DEFINED CUDA_TOOLKIT_ROOT_DIR)
```

Comment:
Is `CUDA_TOOLKIT_ROOT_DIR` strictly necessary for cuda support?

Or could/should it be optional, and if it is present, we add `${CUDA_TOOLKIT_ROOT_DIR}/lib[64]` ?

I.e. was this always required, and this PR is adding a new check, or is this a new requirement?

Reply:
For `find_package()` the way it is currently used, it is required. 

We are also relying on this for `clang-cuda`. 

I believe we were always requiring this, but not explicitly stating it.

That said, I think all this is now deprecated and no longer required. However, we seem to be stuck in a hybrid state where we use both the deprecated stuff and new CUDA Language support available with CMake, which gets confusing.

Reply:
I guess if it is not broken, don't fix it....But, I'll submit an issue on this and when the opportunity arises this can be revamped quite a bit...

Reply:
See #360